app/src/main/java/ch/hikemate/app/model/authentication/AuthRepository.kt

```diff
@@ -71,4 +72,29 @@ interface AuthRepository {
    * @param onSuccess Callback to invoke after the user has successfully signed out.
    */
   fun signOut(onSuccess: () -> Unit = {})
+
+  /**
+   * Deletes the current user's account and invokes the success callback. Takes the user's email and
+   * password as parameters for account deletion as we need to re-authenticate the user before
+   * deleting the account.
+   *
+   * @param password The password of the user.
+   * @param activity The Android Activity, used for launching the re-authentication dialog.
+   * @param onSuccess Callback to invoke after the user's account has been successfully deleted.
+   * @param onErrorAction Callback to invoke when an error occurs during account deletion. Passes
+   *   the Throwable error.
+   */
+  fun deleteAccount(
+      password: String,
+      activity: Activity,
+      onSuccess: () -> Unit = {},
+      onErrorAction: (Exception) -> Unit
+  )
+
+  /**
+   * Return true if the user is signed in with an email provider.
+   *
+   * @param user The FirebaseUser object.
+   */
+  fun isEmailProvider(user: FirebaseUser): Boolean
```

Comment:
Nitpick: Sorry about this, really nitpicking. Could you elaborate about what is an "email provider"? Strictly speaking, Facebook Auth provides email, so does Google, and so on... It might be clear when you're doing this, but for a newcomer, it's not easily understandable